src/lib/WIFI/devWIFI.cpp

```diff
@@ -796,6 +801,238 @@ static void WebUploadForceUpdateHandler(AsyncWebServerRequest *request) {
+    // Complete upload and set error flag
+    bool uploadError = false;
+    uint8_t param = 0;
+    writeLR1121Bytes(NULL, left_over);
+    totalSize += left_over;
+
+    SPIEx.setHwCs(true);
+    if (GPIO_PIN_NSS_2 != UNDEF_PIN)
+    {
+        spiAttachSS(SPIEx.bus(), 1, GPIO_PIN_NSS_2);
+    }
+
+    if (totalSize == expectedFilesize)
+    {
+        DBGLN("reboot 1121");
+        uint8_t reboot_cmd[] = {
+            (uint8_t)(LR11XX_BL_REBOOT_OC >> 8),
+            (uint8_t)LR11XX_BL_REBOOT_OC,
+            0
+        };
+        SPIEx.write(updatingRadio, reboot_cmd, 3);
+        while(!hal.WaitOnBusy(updatingRadio));
+
+        DBGLN("check not in BL mode");
+        packet[0] = LR11XX_SYSTEM_GET_VERSION_OC >> 8;
+        packet[1] = (uint8_t)LR11XX_SYSTEM_GET_VERSION_OC;
+        SPIEx.write(updatingRadio, packet, 2);
+        hal.WaitOnBusy(updatingRadio);
+        memset(packet, 0, sizeof(packet));
+        SPIEx.read(updatingRadio, packet, 5);
+        hal.WaitOnBusy(updatingRadio);
+        uploadError = (packet[2] != 3);
+        DBGLN("hardware %x", packet[1]);
+        DBGLN("type %x", packet[2]);
+        DBGLN("firmware %x", ( ( uint16_t )packet[3] << 8 ) + ( uint16_t )packet[4]);
+    }
+
+    String msg;
+    if (!uploadError && totalSize == expectedFilesize) {
+        msg = String(R"({"status": "ok", "msg": "Update complete. Refresh page to see new version information."})");
+        DBGLN("Update complete");
+    } else {
+        StreamString p = StreamString();
+        if (totalSize != expectedFilesize) {
+            p.println("Not enough data uploaded!");
+        } else {
+            p.println("Update failed, refresh and try again.");
+        }
+        DBGLN("Failed to upload firmware: %s", p.c_str());
+        msg = String(R"({"status": "error", "msg": ")") + p + R"("})";
+    }
+    AsyncWebServerResponse *response = request->beginResponse(200, "application/json", msg);
+    response->addHeader("Connection", "close");
+    request->send(response);
+    request->client()->close();
+}
+
+static void WebUploadLR1121DataHandler(AsyncWebServerRequest *request, const String& filename, size_t index, uint8_t *data, size_t len, bool final) {
+    if (index == 0) {
+#ifdef HAS_WIFI_JOYSTICK
+        WifiJoystick::StopJoystickService();
+#endif
+        expectedFilesize = request->header("X-FileSize").toInt();
+        updatingRadio = request->header("X-Radio").toInt();
+        DBGLN("Update: '%s' size %u on radio %d", filename.c_str(), expectedFilesize, updatingRadio);
+        totalSize = 0;
+
+        // Reboot to BL mode
+        DBGLN("Reboot 1121 to bootloader mode");
+        uint8_t reboot_cmd[] = {
+            (uint8_t)(LR11XX_SYSTEM_REBOOT_OC >> 8),
+            (uint8_t)LR11XX_SYSTEM_REBOOT_OC,
+            3
+        };
+        SPIEx.write(updatingRadio, reboot_cmd, 3);
+        while(!hal.WaitOnBusy(updatingRadio)) {
+            DBGLN("Waiting...");
+            delay(10);
+        }
```

Comment:
The reboot and version query code between this an the one above are _almost_ the same? Can these not be combined to have the same behavior instead of slightly different code copy pasted?